src/Npgsql/NpgsqlConnection.cs

```diff
@@ -135,7 +135,10 @@ public INpgsqlTypeMapper TypeMapper
         /// <see cref="NpgsqlConnection">NpgsqlConnection</see> class.
         /// </summary>
         public NpgsqlConnection()
-            => GC.SuppressFinalize(this);
+        {
+            GC.SuppressFinalize(this);
+            Settings = new NpgsqlConnectionStringBuilder();
```

Comment:
It's mostly the same as what you proposed in #1844. A new instance is allocated each time when a `NpgsqlConnection` is constructed, but it's unused and will be replaced later by setting the `ConnectionString` property.